### DISCUSSION TITLE
Add onLogin & onLogout hooks

### DISCUSSION THREAD

**Discussion by fabhed** [nuxt-community/auth-module#1595] [Ideas & Feature Requests]
### What problem does this feature solve?

It allows for users to add custom functionality in one place.

### What does the proposed changes look like?

```js
export default function({ $auth }) {
  $auth.onLogin((strategy, data) => {
    // Do stuff
  })

  $auth.onLogout(() => {
    // Do stuff
  })
}
```

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/auth-module/issues/c616">#c616</a>)</em></sub></div>

**Comment by NickBolles** [comment]
you can watch Logged in -> https://auth.nuxtjs.org/api/storage.html#local-state

I did find an issue with this callback not being `Awaited` though. So be warned about that. If this is implemented, implenting it as an awaitable stack would be awesome (or even something like [KOAs middleware cascading](https://koajs.com/#application))
